src/mlpack/methods/ann/layer/layer_norm.hpp

```diff
@@ -0,0 +1,197 @@
+/**
+ * @file layer_norm.hpp
+ * @author Shikhar Jaiswal
+ *
+ * Definition of the Layer Normalization class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_LAYERNORM_HPP
+#define MLPACK_METHODS_ANN_LAYER_LAYERNORM_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * Declaration of the Layer Normalization class. The layer transforms
+ * the input data into zero mean and unit variance and then scales and shifts
+ * the data by parameters, gamma and beta respectively over a single training
+ * data. These parameters are learnt by the network.
```

Comment:
If you like, it might be useful to add a comment here differentiating 'layer normalization' from 'batch normalization'---I imagine many users will be familiar with the terms, but not have any insight on why to use which one.  So if you can think of something to add here, it could be useful, otherwise I think it is not a huge deal. :)